File: 1482-HowManyNumbersAreSmallerThanTheCurrentNumber/1482-HowManyNumbersAreSmallerThanTheCurrentNumber.py
# Last updated: 4/14/2026, 12:48:25 PM
class Solution(object):
    def smallerNumbersThanCurrent(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        temp = sorted(nums)
        d = {}
        ret = []
        for i , v in enumerate(temp):
            if v not in d:
                d[v]=i
        
        for i in nums:
            ret.append(d[i])
        return ret

File: 1482-HowManyNumbersAreSmallerThanTheCurrentNumber/1482-HowManyNumbersAreSmallerThanTheCurrentNumber.js
// Last updated: 4/14/2026, 12:48:20 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
    let temp = [...nums].sort((a,b)=>a-b)
    let n = temp.length
    let d = {}

    for(let i = 0 ; i <n ; i++){
        if(!(temp[i] in d)){
            d[temp[i]]=i
        }
    }
    let ret = []

    for(let num of nums){
        ret.push(d[num])
    }return ret
};